data-processing/entities/definitions/model/configuration.py

```diff
@@ -113,3 +113,7 @@ class DataTrainingArguments:
     max_seq_len: Optional[int] = field(
         default=50, metadata={"help": "TBW"},
     )
+    dataconfig_file: Optional[str] = field(
+        default="dataconfig.json", metadata={"help": "dataconfig file (it has intent_label, slot_label dicts, and pos_label list)"},
```

Comment:
Should there be a dataconfig.json file as part of this commit?

Also, what is the expected format of the dataconfig.json file? If it is a standard file format (e.g., documented as part of the Pytorch library or something), then can a link be provided here?

Reply:
No, It's bundled along with the model. Earlier, we had `intent_label.txt` , `pos_label.txt` and `slot_label.txt` files bundled with each model. But since this model was trained on 3 datasets, we would have had to have 9 files with names corrected. So, I made a json file with all the content for better maintenance.  Where do you think is a good place to describe it's schema?

Reply:
Are there any circumstnades under which we would use something besides "dataconfig.json" as the dataconfig file? If not, then perhaps we should just code this as a constant.

Reply:
> So, I made a json file with all the content for better maintenance. Where do you think is a good place to describe it's schema?

I would recommend two things:

1. Providing a link to one of the dataconfig.json file on S3 (someone can open this file to see the format themselves)
2. Add documentation of the schema right here as comments (in particular, what are the fields and allowable values)

Reply:
It is actually in the zip of the model 😓 . But I shall add a note and the schema

Reply:
Done!